xl2times/transforms.py

```diff
@@ -2227,78 +2321,19 @@ def apply_transform_tables(
-        qs = []
-        if process is not None:
-            qs.append(f"process in ['{process}']")
-        if commodity is not None:
-            qs.append(f"commodity in ['{commodity}']")
-        if attribute is not None:
-            qs.append(f"attribute == '{attribute}'")
-        if region is not None:
-            qs.append(f"region == '{region}'")
-        if year is not None:
-            qs.append(f"year == {year}")
-        return table.query(" and ".join(qs)).index
-
-    def eval_and_update(
-        table: DataFrame, rows_to_update: pd.Index, new_value: str
-    ) -> None:
-        """Performs an inplace update of rows `rows_to_update` of `table` with `new_value`,
-        which can be a update formula like `*2.3`."""
-        if isinstance(new_value, str) and new_value[0] in {"*", "+", "-", "/"}:
-            old_values = table.loc[rows_to_update, "value"]
-            updated = old_values.astype(float).map(lambda x: eval("x" + new_value))
-            table.loc[rows_to_update, "value"] = updated
-        else:
-            table.loc[rows_to_update, "value"] = new_value
-
     if datatypes.Tag.tfm_upd in tables:
         updates = tables[datatypes.Tag.tfm_upd]
         table = tables[datatypes.Tag.fi_t]
-        new_tables = [table]
+
         # Reset FI_T index so that queries can determine unique rows to update
         tables[datatypes.Tag.fi_t].reset_index(inplace=True, drop=True)
 
-        # TFM_UPD: expand wildcards in each row, query FI_T to find matching rows,
-        # evaluate the update formula, and add new rows to FI_T
-        # TODO perf: collect all updates and go through FI_T only once?
-        for _, row in tqdm(
-            updates.iterrows(),
-            total=len(updates),
-            desc=f"Applying transformations from {datatypes.Tag.tfm_upd.value}",
-        ):
-            rows_to_update = query(
-                table,
-                row["process"],
-                row["commodity"],
-                row["attribute"],
-                row["region"],
-                row["year"],
-            )
+        # Apply the update formulas in updates to the rows in table
+        new_tables = _eval_updates(table, updates)
 
-            if not any(rows_to_update):
-                logger.info(
-                    f"A {datatypes.Tag.tfm_upd.value} row generated no records."
-                )
-                continue
-
-            new_rows = table.loc[rows_to_update].copy()
-            new_rows["source_filename"] = row["source_filename"]
-            eval_and_update(new_rows, rows_to_update, row["value"])
-            new_tables.append(new_rows)
-
-        # Add new rows to table
-        tables[datatypes.Tag.fi_t] = pd.concat(new_tables, ignore_index=True)
+        # Append updated rows to table.
+        # TODO this matches old behaviour, but double-check that the old (non-updated) values get removed later?!
+        tables[datatypes.Tag.fi_t] = pd.concat([table, new_tables], ignore_index=True)
```

Comment:
There is a pandas depreciation warning when new_tables is empty.

Reply:
Interesting, I don't see that.  What pandas version are you on?

ps. this is where poetry would be handy, as it forces us (because of its lock file) to all test with the exact same lib versions, rather than the current ambiguous `pandas >=2.1` range.

Reply:
The warning normally appears on the CI, I think. I've been using version 2.2.